examples/cloud-run/main.tf

```
@@ -0,0 +1,162 @@
+}
+
+data "archive_file" "src" {
+  type        = "zip"
+  source_dir  = "./cloudrun"
+  output_path = "cloudrun.zip"
+}
+
+resource "google_storage_bucket_object" "object" {
+  name   = "pub-sub-function-source.zip"
+  bucket = google_storage_bucket.bucket.name
+  source = data.archive_file.src.output_path # Add path to the zipped function source code
+}
+
+resource "google_service_account" "gcf_sa" {
+  project      = var.project_id
+  account_id   = "gcf-sa"
+  display_name = "Test Service Account"
+}
+
+resource "google_project_iam_member" "gcf_invoker_role" {
+  project = var.project_id
+  role    = "roles/run.invoker"
+  member  = "serviceAccount:${google_service_account.gcf_sa.email}"
+}
+
+module "cloud-function" {
+  source  = "GoogleCloudPlatform/cloud-functions/google"
+  version = "~> 0.6"
+
+  function_name     = "secret-manager-pub-sub-consumer"
+  project_id        = var.project_id
+  function_location = var.region
+  runtime           = "python312"
+  entrypoint        = "subscribe"
+  storage_source = {
+    bucket = google_storage_bucket.bucket.name
+    object = google_storage_bucket_object.object.name
+  }
+
+  event_trigger = {
+    trigger_region        = var.region
+    event_type            = "google.cloud.pubsub.topic.v1.messagePublished"
+    pubsub_topic          = google_pubsub_topic.secret_topic.id
+    retry_policy          = "RETRY_POLICY_RETRY"
+    service_account_email = google_service_account.gcf_sa.email
+  }
+
+  service_config = {
+    max_instance_count               = 1
+    min_instance_count               = 1
+    available_memory                 = "256M"
+    timeout_seconds                  = 60
+    max_instance_request_concurrency = 80
+    available_cpu                    = "1"
+    ingress_settings                 = "ALLOW_INTERNAL_ONLY"
+    all_traffic_on_latest_revision   = true
+    service_account_email            = google_service_account.gcf_sa.email
+  }
+
+  depends_on = [
+    google_project_iam_member.gcf_invoker_role
+  ]
+}
+
+module "kms" {
+  source  = "terraform-google-modules/kms/google"
+  version = "3.0.0"
+
+  keyring              = "${var.keyring}-${random_id.random_topic_id_suffix.hex}"
+  location             = var.region
+  project_id           = var.project_id
+  keys                 = [local.key_name]
+  purpose              = "ENCRYPT_DECRYPT"
+  key_protection_level = "HSM"
+  prevent_destroy      = false
+}
+
+resource "google_kms_crypto_key_iam_member" "sm_sa_encrypter_decrypter" {
+  role          = "roles/cloudkms.cryptoKeyEncrypterDecrypter"
+  member        = "serviceAccount:${google_project_service_identity.secretmanager_identity.email}"
+  crypto_key_id = module.kms.keys[local.key_name]
+}
+
+module "secret-manager" {
+  source  = "GoogleCloudPlatform/secret-manager/google"
+  version = "~> 0.4"
+
+  project_id = var.project_id
+  secrets = [
+    {
+      name        = "secret-cloud-run-1"
+      secret_data = "secret information"
```

Just an idea, feel free to implement it or not, but how about put more focus on where the sensitive text is?

```suggestion
      secret_data = "SENSITIVE TEXT HERE!"
```